src/Renderer/Device/VulkanDevice.cpp

```diff
@@ -0,0 +1,362 @@
+#include "VulkanDevice.h"
+
+// std headers
+#include <cstring>
+#include <iostream>
+#include <set>
+
+namespace SnekVk {
+
+	VulkanDevice::VulkanDevice(Window &window) : window{window} 
+	{
+		CreateInstance();
+		SetupDebugMessenger();
+		CreateSurface();
+		PickPhysicalDevice();
+		CreateLogicalDevice();
+		CreateCommandPool();
+	}
+
+	VulkanDevice::~VulkanDevice() {}
+
+	void VulkanDevice::CreateInstance() 
+	{
+		if (enableValidationLayers)
+		{
+			SNEK_ASSERT(SnekVk::CheckValidationLayerSupport(validationLayers.data(), validationLayers.size()),
+				"Validation Layers are not supported!");
+		}
+			
+		VkApplicationInfo appInfo = {};
```

Comment:
```suggestion
		VkApplicationInfo appInfo {};
```
for your consideration et. al